### PR TITLE
Add release process to the docs

### PR DESCRIPTION
**Pull request by humblec**
The plan is to have first draft of the release with this doc. We will add more details while we progress.

Refer # https://github.com/ceph/ceph-csi/issues/353

Signed-off-by: hchiramm <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@rootfs @leseb @Madhu-1 @ShyamsundarR ptal

File: docs/releases.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  the previous release* (may be backward incompatible).
+
+## Tagging repositories
+
+The tag name must begin with "v" followed by the version number, conforming to
+the [versioning](#versioning) requirements (e.g. a tag of `v1.0.0-rc2` for
+version `1.0.0-rc2`). This tag format is used by the Travis CI infrastructure to
+properly upload and tag releases to Quay and Docker Hub.
+
+## Release process
```
Comment (ShyamsundarR) [inline docs/releases.md]: Comments for this section:
- Release checklist is a list of issues/PRs that are to be completed before the release is made?
- How can a user relate to a release and figure out what its contents are? IOW, release notes or how to determine an issue or a PR is part of a release?

File: docs/releases.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+The Release Owner must follow the following process, which is
+designed to ensure clarity, quality, stability, and auditability
+of each release:
+
+- Raise an issue and assign to themselves.
+
+[new issue in this repository](https://github.com/Ceph/Ceph-csi-driver/issues/new)
```
Comment (ShyamsundarR) [inline docs/releases.md]: Link needs correction

Comment (humblec) [reply docs/releases.md]: Sure, done.

File: docs/releases.md (hunk 1 above)
Comment (humblec) [reply docs/releases.md]: I have added some more details to address above queries. PTAL.

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR PTAL , I have addressed the comments.

File: docs/releases.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+- [Ceph CSI driver Release Process](#Ceph-csi-driver-release-process)
+  - [Introduction](#introduction)
+  - [Versioning](#versioning)
+  - [Tagging repositories](#tagging-repositories)
+  - [Release process](#release-process)
+
+## Introduction
+
+This document provides details about Ceph CSI driver releases.
```
Comment (ShyamsundarR) [inline docs/releases.md]: Reword releases to release process

File: docs/releases.md (hunk 1 above)
Comment (ShyamsundarR) [reply docs/releases.md]: This is still inadequate, or I am unable to understand how things will move.

As this PR intends to start a release process doc. and update its contents as we progress towards a release, I would suggest adding a TBD here around the various bullets and as time progresses updating the same.

Comment (humblec) [reply docs/releases.md]: Added `TBD` to the section, so it convey it can be imrpoved any time.

File: docs/releases.md (hunk 3 above)
Comment (humblec) [reply docs/releases.md]: Done.

**Comment by humblec** [comment]
@ShyamsundarR comments are addressed. PTAL .

**Comment by humblec** [comment]
@Madhu-1 Can you review this please?

**Pull request merged**